VAMobile/src/components/VAIcon/VAIcon.tsx

```diff
@@ -223,12 +226,24 @@ const VAIcon: FC<VAIconProps> = ({ name, width, height, fill, fill2, stroke, pre
     return <></>
   }
 
-  if (width && isFinite(width)) {
-    iconProps = Object.assign({}, iconProps, { width: preventScaling ? width : fs(width) })
-  }
+  if (width && isFinite(width) && height && isFinite(height)) {
```

Comment:
1. Any idea why isFinite was being used? At face value removing it seemed fine
2. This condition doesn't work correctly as is--does nothing if you only tried to set width or height and not both (which I assume breaks existing behavior somewhere)

I think we might be trying to overcomplicate this, couldn't we just:
```
const truncateWidth = width && fs(width) > maxWidth
// Mostly same as old code
if (width) {
  iconProps = { ...iconProps, width: preventScaling ? width : truncateWidth ? maxWidth : fs(width) }}
}

if (height && truncateWidth) {
  const scaledHeight = ( maxWidth / width ) * height
  iconProps = { ...iconProps, { height: preventScaling ? height : scaledHeight }}
} else if (height) {
  iconProps = { ...iconProps, { height: preventScaling ? height : fs(height) }}
}
```

Maybe could be cleaned up more, but fundamentally: the only real special handling needed is the height in the case we've truncated width.

Reply:
I looked through our existing uses of VAIcon, and found only two cases:
- Both width and height specified
- No dimensions specified

I tried specifying only width or only height to VAIcon to see what happens, and it's not good. The icon file itself specifies the missing dimension so you get an awkward mix of what you specified on one dimension and what the icon file specified on the other dimension. From what I can tell we don't ever actually want that. In order to correctly calculate the missing dimension we'd have to read the file to determine the aspect ratio.

I don't think the isFinite check is needed either. I assume it was added originally to ensure people didn't try to specify values like '22px' but since we have a TS checking the type, that should be sufficient to warn people that they didn't provide a number. I removed the isFinite check.